src/lib/DepositController.js

```diff
@@ -68,11 +81,50 @@ export class DepositController {
       });
 
       formik.setSubmitting(false);
-
       const recordURL = response.data.links.self_html;
       window.location.replace(recordURL);
     } catch (error) {
       store.dispatch(setFormErrorsFromResponse(error, formik));
     }
   }
+
+  uploadRecordFiles = async (record, files, { store }) => {
+    const recordSerializer = store.config.recordSerializer;
+    let payload = recordSerializer.serialize(record);
+    //Create draft
+    if (!this.exists(payload)) {
+      const response = await this.apiClient.create(payload);
+      store.dispatch({
+        type: CREATE_DEPOSIT_SUCCESS,
+        payload: { data: recordSerializer.deserialize(response.data) },
+      });
+      const newURL = response.data.links.self_html;
+      window.history.replaceState(undefined, '', newURL);
+    }
+    store.dispatch({
+      type: 'FILE_UPLOAD_INITIATE',
+      payload: files.map((file) => ({ fileName: file.name, size: file.size })),
+    });
+
+    for (const file of files) {
+      this.fileUploader.upload(file, { store });
+    }
+  };
+
+  async deleteRecordFile(file, { formik, store }) {
+    //Create draft
+    const fileVersionUrl = file.links.version;
+    try {
+      const resp = await this.apiClient.deleteFile(fileVersionUrl);
+      store.dispatch({
+        type: 'FILE_DELETED_SUCCESS',
+        payload: {
+          fileName: file.fileName,
+        },
+      });
+    } catch (e) {
+      console.log('error');
```

Comment:
I guess error reporting to user will be improved...

Reply:
Yes it's far from finished that part :)